src/test/java/cz/muni/fi/pa165/library/repositories/BookRepositoryTest.java

```diff
@@ -1,42 +1,87 @@
 package cz.muni.fi.pa165.library.repositories;
 
 import cz.muni.fi.pa165.library.entities.Book;
-import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
-import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.dao.InvalidDataAccessApiUsageException;
 import org.springframework.test.context.junit4.SpringRunner;
-
+import java.util.Arrays;
 import java.util.List;
 
-import static org.hamcrest.MatcherAssert.assertThat;
-
+/**
+ * @author Katarína Hermanová
+ * UČO 433511
+ * Github katHermanova
+ */
 @RunWith(SpringRunner.class)
 @DataJpaTest
 public class BookRepositoryTest {
 
-    @Autowired
-    private TestEntityManager entityManager;
-
     @Autowired
     private BookRepository bookRepository;
+    private Book book1;
+    private Book book2;
 
-    @Test
-    public void findByEmail() {
-        Book book1 = new Book();
+    @Before
+    public void setUp() {
+        book1 = new Book();
```

Comment:
I understand the intention not to have to recreate sample book entities each time, but creating it in each test method adds a lot to readability. Also you have global mutable objects common to multiple tests, which is error prone. I would prefer if you created some private method like createSampleBook which would return the book you need.